### PR TITLE
Quality reinforcement (6): fix unstable test cases

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
fix unstable test cases in CI.

- [x] ha_cases test_multi_task_reduce_and_restart_worker, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7059/pipeline), added log, GMHDBJD
- [x] ha_cases test_multi_task_running, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7273/pipeline/77), [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7273/pipeline/), [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7276/pipeline/), [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7242/pipeline), syncer runs slowly, no error, https://github.com/pingcap/dm/pull/1153/commits/af043212c10c4ee8503961bb1e48def174631423, lance6716
- [x] ~ha_cases [network problem?](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7124/pipeline)~
- [x] ha_cases [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7005/pipeline), https://github.com/pingcap/dm/issues/1110
- [x] ha_cases test_pause_task [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7185/pipeline), [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7212/pipeline/77), [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7217/pipeline), https://github.com/pingcap/dm/pull/1153/commits/e80e24b2b644af24c149d7dde78f638a2559c262, lance6716
- [x] testRelaySuite.TestProcess, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7194/pipeline/68), https://github.com/pingcap/dm/pull/1153/commits/5d93432d30ca90acce28aa9e844e7f8ea775a52a, lance6716
- [x] ~testPessimist.TestUnlockSourceInterrupt, [outdated](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6806/pipeline)~
- [ ] testMaster.TestOfflineMember: [outdated](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6814/pipeline), [outdated](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6939/pipeline/) what to do if leader forwarding fails?
- [x] testPessimist.TestPessimist: [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6947/pipeline), https://github.com/pingcap/dm/pull/1153/commits/4355bcb601edce2a8e370f478086f3a91e2aa8c7, lance6716
- [x] ~shardddl/optimism [outdated](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/6992/pipeline)~
- [x] ~testScheduler.TestRestartScheduler, [outdated](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7046/pipeline)~
- [x] ~DM-CASE-94, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7090/), didn't finish dump (why slowly)?~
- [x] ~testForEtcd.TestOperationEtcd, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7093/pipeline/), seems timeout~
- [ ] ha_master test_evict_leader, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7165/pipeline), lance6716, #1214 (but root cause not found)
- [x] DM-RemoveLock, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7168/pipeline), sleep 5 second is not enough when run slowly, https://github.com/pingcap/dm/pull/1153/commits/be85c87c934f73129a4a9b6fc00713d8d34a0473, lance6716
- [x] DM-102, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7272/pipeline/91), https://github.com/pingcap/dm/pull/1153/commits/25429e6932ab39bb50c3a1fc887fca01d2874bc4, lance6716
- [ ] optimism.TestInfo deadlock, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7315/pipeline), no idea, add log https://github.com/pingcap/dm/pull/1153/commits/3676df73ba2e2c29216502d1ae6c28b5186b9a21
- [x] ~ha, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7334/pipeline), dmctl dail DM-master timeout~
- [ ] DM-078, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7369/pipeline), can't connect to worker and worker log has been cleaned?
- [x] ~ha_cases test_multi_task_running, [jenkins](https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/7399/pipeline), fixed by not merged into that PR~

### CODE REVIEW

**Comment by lance6716** [comment]
will disable stress now, seems our daily PR could reveal enough unstable CI (more than we could fix)

**Comment by csuzhangxc** [comment]
/run-all-tests

for coverage

File: tests/shardddl3/run.sh [hunk 1]
```diff
@@ -610,6 +610,23 @@ function DM_102_CASE() {
...
 
+    sleep 1
+    # wait DM receive source2's DDL
+    found=false
+    for ((k=0; k<10; k++)); do
+        content=$($PWD/bin/dmctl.test DEVEL --master-addr=127.0.0.1:$MASTER_PORT query-status test)
+        master2=$(echo $content | sed 's/"masterBinlog":/"masterBinlog":\n/g' | awk -F')' 'FNR==3{print $1}')
+        syncer2=$(echo $content | sed 's/"syncerBinlog":/"syncerBinlog":\n/g' | awk -F')' 'FNR==3{print $1}')
+        if [ "$master2" != "$syncer2" ]; then
+            found=true
```
Comment (csuzhangxc) [inline tests/shardddl3/run.sh:621]: Is there other cause can lead to `"$master2" != "$syncer2"`?

Comment (lance6716) [reply tests/shardddl3/run.sh:621]: I guess no. syncer was initiated at the position where no modification made in upstream, at then `"$master2" == "$syncer2"`. and only modification after that is source2's DDL which we are waiting

**Review by csuzhangxc — COMMENTED** [review]
LGTM

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
/run-all-tests

etcd timeout

**Pull request merged**